.github/workflows/build_and_test.yml

```diff
@@ -828,7 +821,7 @@ jobs:
         if: ${{ needs.tag.outputs.build-tag == github.run_id }}
```

Comment:
This condition reads horribly. What this means is "we're in a PR", but there's better ways of figuring this out.

I wanted to adjust this so that it doesn't run on proxy/storage releases because it doesn't matter there, and nearly missed this doesn't run on any releases anyway.

Reply:
> Do I read this correctly as "we're in a PR"? 

Yeah, sounds about right, but also not in the release PR. So probably something like `if: github.event_name == 'pull_request' && !startsWith(github.base_name, 'release')`

Reply:
good catch! Keeping the conversation open so that I remember to amend this tomorrow.

Reply:
reading up on this again, this seems to not be correct.

https://github.com/neondatabase/neon/actions/runs/13331343751/job/37236906599 ran on https://github.com/neondatabase/neon/pull/10820, a storage release, and got a `run-id` based `build-tag`